scaladoc/resources/dotty_res/scripts/theme.js

```diff
@@ -0,0 +1,33 @@
+;(function () {
+  const supportsLocalStorage = (() => {
+    try {
+      localStorage.setItem('test', 'test');
+      localStorage.removeItem('test');
+      return true;
+    } catch (e) {
+      return false;
+    }
+  })();
+
+  function toggleDarkTheme(isDark) {
+    currentlyDark = isDark
+    // this triggers the `:root.theme-dark` rule from scalastyle.css,
+    // which changes the values of a bunch of CSS color variables
+    document.documentElement.classList.toggle("theme-dark", isDark);
+    supportsLocalStorage && localStorage.setItem("use-dark-theme", isDark);
+  }
+
+  /* This needs to happen ASAP so we don't get a FOUC of bright colors before the dark theme is applied */
+  const initiallyDark = supportsLocalStorage && (localStorage.getItem("use-dark-theme") === "true");
```

Comment:
If not theme has been explicitly selected by the user, the default should be dark if the system theme is dark, in CSS this can be queried using prefers-color-scheme: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

Reply:
looks like `window.matchMedia('(prefers-color-scheme: dark)')` should do the trick in JS.

Reply:
Ooooh that's pretty cool, I didn't know about that feature. I just pushed a commit to add that logic. I wired it up via JS as opposed to CSS since it needs to interact with the switch, and I think a CSS-only solution would require me to duplicate all of the variable settings (one for each media query).

Reply:
this is really nice!